typing/ctype.ml

```diff
@@ -7281,6 +7281,31 @@ let print_global_state fmt global_state =
 let type_equal env ty1 ty2 = is_equal env false [ty1] [ty2]
 let () = type_equal' := type_equal
 
+let compute_variant_with_null_jkind env type_jkind_purely ty =
+  match get_desc ty with
+  | Tconstr (path, args, _) ->
+    begin try
+      let decl = Env.find_type path env in
+      match decl.type_kind with
+      | Type_variant ([_; {cd_args = Cstr_tuple [{ ca_type; ca_modalities }];
+         _}], Variant_with_null, _) ->
```

Comment:
maybe it's worth adding `extract_or_null_variant` or something like that to `Types`? This seems a little specific to be nesting in such a dark and dusty corner

Reply:
Yeah, good idea.